Modules/_threadmodule.c

```diff
@@ -1219,7 +1219,8 @@ rlock_repr(PyObject *op)
 {
     rlockobject *self = rlockobject_CAST(op);
     PyThread_ident_t owner = self->lock.thread;
-    size_t count = self->lock.level + 1;
+    size_t count = _PyRecursiveMutex_IsLockedByCurrentThread(&self->lock) ?
```

Comment:
This changes the output of `repr(rlock)` on an `rlock` locked by another thread. Previously, it would print the lock level; now it prints zero.

Is that intentional?

Reply:
> Is that intentional?

No, it's an error. Test should be: `PyMutex_IsLocked(self->lock.mutex)`.